packages/pds/src/api/com/atproto/sync/getRecord.ts

```diff
@@ -20,17 +20,22 @@ export default function (server: Server, ctx: AppContext) {
           throw new InvalidRequestError(`Could not find repo for DID: ${did}`)
         }
       }
-      const storage = new SqlRepoStorage(ctx.db, did)
+      // must open up the db outside of store interface so that we can close the file handle after finished streaming
+      const actorDb = await ctx.actorStore.db(did)
+      const storage = new SqlRepoReader(actorDb)
       const commit = params.commit
         ? CID.parse(params.commit)
         : await storage.getRoot()
+
       if (!commit) {
         throw new InvalidRequestError(`Could not find repo for DID: ${did}`)
       }
-      const proof = repo.getRecords(storage, commit, [{ collection, rkey }])
+      const carIter = repo.getRecords(storage, commit, [{ collection, rkey }])
+      const carStream = byteIterableToStream(carIter)
+      carStream.on('close', actorDb.close)
```

Comment:
I wonder if this db close may not be necessary.  If we keep it around my concern would be that it could stick in the db cache but be closed.